### ISSUE TITLE
False positive in unneeded_synthesized_initializer

### ISSUE THREAD

**Issue by ZevEisenberg** [realm/SwiftLint#5075]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

`unneeded_synthesized_initializer` reports a false positive when the init has side effects. In the attached sample, the `init` prints something, and removing it would remove the print. In the example where I encountered it in a real app, a SwiftUI view modifier's `init` contained a call to `UIScrollView.appearance().bounces = false`, which we definitely don't want to lose by deleting the `init`!

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
Linting Swift files in current working directory
Linting 'InitTest.swift' (1/1)
/Users/igeek/Downloads/InitTest/InitTest.swift:2:5: warning: Unneeded Synthesized Initializer Violation: This default initializer would be synthesized automatically - you do not need to define it (unneeded_synthesized_initializer)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version: **0.52.3**
* Installation method used: **homebrew**
* Paste your configuration file: **n/a (default configuration)**
* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? **No**
* Which Xcode version are you using? **Xcode 14.3.1 (14E300c) on macOS 13.4.1 (22F82)**
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
struct MyStruct {
    init() {
        print("perform side effect")
    }
}
```

**Comment by mildm8nnered** [comment]
interesting - taking a look shortly

**Comment by ZevEisenberg** [comment]
Quick work! Thank you @mildm8nnered!
